php_phongo.c

```diff
@@ -1638,34 +1627,32 @@ static void phongo_result_iterator_move_forward(zend_object_iterator *iter TSRML
 
 			bson_iter_document(&cursor_it->first_batch_iter, &data_len, &data);
 
-			MAKE_STD_ZVAL(result->visitor_data.zchild);
-			bson_to_zval(data, data_len, &result->visitor_data);
+			MAKE_STD_ZVAL(cursor->visitor_data.zchild);
+			bson_to_zval(data, data_len, &cursor->visitor_data);
 			return;
 		}
 	}
-	if (mongoc_cursor_next(result->cursor, &doc)) {
-		MAKE_STD_ZVAL(result->visitor_data.zchild);
-		bson_to_zval(bson_get_data(doc), doc->len, &result->visitor_data);
+	if (mongoc_cursor_next(cursor->cursor, &doc)) {
+		MAKE_STD_ZVAL(cursor->visitor_data.zchild);
+		bson_to_zval(bson_get_data(doc), doc->len, &cursor->visitor_data);
 	} else {
-		iter->funcs->invalidate_current(iter TSRMLS_CC);
+		/* TODO: is this really necessary? */
```

Comment:
Because it'll be called during the dtor?
Maybe its not strictly necessary but I'd prefer to keep it in case someone can find a quirky way of triggering move_forward outside of normal routines.
.. remove the comment?